apps/tutorial-servo-can-control/src/TutorialServo.cpp

```diff
@@ -0,0 +1,43 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "TutorialServo.h"
+
+#include "mbed.h"
+
+// CONSTRUCTOR
+TutorialServo ::TutorialServo(PinName servoPin, float servoRangeInDegrees = 180.0, float minPulsewidthInMs = 1,
+                              float maxPulsewidthInMs = 2) {
+  PwmOut servoPwmOut(servoPin);
+  m_servoPwmOut = servoPwmOut;
+  m_servoRangeInDegrees = servoRangeInDegrees;
+  m_minPulsewidthInMs   = minPulsewidthInMs;
+  m_maxPulsewidthInMs   = maxPulsewidthInMs;
+}
+
+// Set servo position (ex. 45 deg) --> assume servo can't make continuous motion
+void TutorialServo ::setPositionInDegrees(const float degrees) {
+  // check bounds on degrees. modify if out of bounds
+  float newPosition = m_minPulsewidthInMs + m_minPulsewidthInMs * (degrees / m_servoRangeInDegrees);
```

Comment:
Rethink how you're mapping from degrees to pulse width. Say m_minPulsewidthInMs = 2ms and m_maxPulsewidthInMs = 3ms, then if degrees = 90 and m_servoRangeInDegrees = 180, the mapping would give newPosition = 3ms, when it should've been mapped to halfway between min/max pulse widths.